src/api/system.c

```diff
@@ -109,9 +109,14 @@ static const char *get_key_name(const SDL_Event *e, char *buf) {
     /* We need to correctly handle non-standard layouts such as dvorak.
        Therefore, if a Latin letter(code<128) is pressed in the current layout,
        then we transmit it as it is. But we also need to support shortcuts in
-       other languages, so for non-Latin characters we pass the scancode that
-       matches the letter in the QWERTY layout. */
-    if (e->key.keysym.sym < 128)
+       other languages, so for non-Latin characters(code>128) we pass the
+       scancode based name that matches the letter in the QWERTY layout.
+       
+       In SDL, the codes of all special buttons such as control, shift, arrows
+       and others, are shifted using bit shift 1<<30(SDLK_SCANCODE_MASK) outside
+       the unicode range(>0x10FFFF). User can remap these buttons, so we need
+       to give correct name, not scancode based.*/
+    if ((e->key.keysym.sym < 128) || (e->key.keysym.sym > SDLK_SCANCODE_MASK))
```

Comment:
Doing this comparison is probably fine, but maybe the correct one is `e->key.keysym.sym & SDLK_SCANCODE_MASK`.

Reply:
Yeah; that's probably "more" correct, but as you say, either should work.

Reply:
I assume you meant: `(e->key.keysym.sym & ~SDLK_SCANCODE_MASK) < 128`
It won't work. Special key scancodes do not lie in a certain range in order. Here's a list: 
https://github.com/libsdl-org/SDL/blob/c59d4dcd38c382a1e9b69b053756f1139a861574/include/SDL_scancode.h

I think they were deliberately shifted so that they could be conveniently checked.

Reply:
No; I think exactly Guldoman's operation; because .sym is going to be a result of SDL_SCANCODE_TO_KEYCODE, which is basically (x | (1<<30)), theoretically just checking with `&` should work, no?

Reply:
Nope, I really meant `e->key.keysym.sym & SDLK_SCANCODE_MASK`.
SDL calls this macro
```C
#define SDLK_SCANCODE_MASK (1<<30)
#define SDL_SCANCODE_TO_KEYCODE(X)  (X | SDLK_SCANCODE_MASK)
```
which just sets the 30th bit to `1`.
Doing `e->key.keysym.sym & SDLK_SCANCODE_MASK` checks if the bit is `1`.

Reply:
Understood what you mean. I don't see the difference though.

Reply:
Yeah; there's not any, I don't think. The only concern would be if there's a key > 1<<30, but that doesn't actually have the 30th bit set. Not sure if that exists; I doubt it. I'd honestly accept the PR as-is.

Reply:
It works the same and my initial comment was just a nit.

Reply:
There is a set of special buttons, the scancodes of which are in the range from 0 to 512. Sdl, when converting to keycodes, shifts this set of scancodes from 0 to 2^30. This is the same as saying that it changes the 30th bit. Or that a bitwise OR operation with a mask of 1<<30 is applied. I don't see the difference between all of this.
Your English is better than mine. Write whatever you want. You seem to have rights.